### DISCUSSION TITLE
monai.transforms.RandAffined

### DISCUSSION THREAD

**Discussion by agokli** [Project-MONAI/MONAI#8028] [General]
I am new to MONAI and trying to understand class monai.transforms.RandAffined. I have seen several tutorials using this specific transform. I still do not understand parameters. For an example, rand_affine = RandAffined(
    keys=["image", "label"],
    mode=("bilinear", "nearest"),
    prob=1.0,
    spatial_size=(300, 300, 50),
    translate_range=(40, 40, 2),
    rotate_range=(np.pi / 36, np.pi / 36, np.pi / 4),
    scale_range=(0.15, 0.15, 0.15),
    padding_mode="border",
)
(a) I am assuming that spatial_size is output batch size. 
(b) How does rotate_range parameters get translated in rotational angles?
(c) How are rotate_range, translate_range and scale_range are correlated
(d) In general, how do you determine these parameters values? Any articles for providing such understanding?

**Comment by KumoLiu** [comment]
Hi @agokli, thanks for your interest here.
Hope you can find some information from the doc-string. The parameter selection often requires experimentation and fine-tuning depending on the specific application, perhaps you can try more experiments and check the output.
https://github.com/Project-MONAI/MONAI/blob/cea80a686b907b22d90c8024cea5a17ef9d9f58a/monai/transforms/spatial/array.py#L2343
